aiida/backends/sqlalchemy/migrations/versions/37f3d4882837_make_all_uuid_columns_unique.py

```diff
@@ -0,0 +1,58 @@
+"""Make all uuid columns unique
+
+Revision ID: 37f3d4882837
+Revises: 041a79fc615f
+Create Date: 2018-11-17 17:18:58.691209
+
+"""
+# pylint: disable=invalid-name
+
+from __future__ import absolute_import
+from alembic import op
+
+# Remove when https://github.com/PyCQA/pylint/issues/1931 is fixed
+# pylint: disable=no-member,no-name-in-module,import-error
+
+# revision identifiers, used by Alembic.
+revision = '37f3d4882837'
+down_revision = '041a79fc615f'
+branch_labels = None
+depends_on = None
+
+tables = ['db_dbcomment', 'db_dbcomputer', 'db_dbgroup', 'db_dbworkflow']
+
+
+def verify_uuid_uniqueness(table):
+    """Check whether the database contains duplicate UUIDS.
+
+    Note that we have to redefine this method from aiida.manage.database.integrity.verify_uuid_uniqueness
+    because that uses the default database connection, while here the one created by Alembic should be used instead.
+
+    :raises: IntegrityError if database contains nodes with duplicate UUIDS.
+    """
+    from sqlalchemy.sql import text
+    from aiida.common.exceptions import IntegrityError
+
+    query = text(
+        'SELECT s.id, s.uuid FROM (SELECT *, COUNT(*) OVER(PARTITION BY uuid) AS c FROM {}) AS s WHERE c > 1'.format(
```

Comment:
This is correct, but a (possibly faster) way to get the information without subqueries and aggregate functions is just using group by, something like
```sql
SELECT uuid, COUNT(uuid) from {} GROUP BY uuid HAVING COUNT(uuid) > 1
```
(that gives a different result, but in the end you just use the result to know if there are duplicates, not which one they are).

Just for info anyway (and possibly you need the aggregate when actually deduplicating the uuid).